File: solutions/3843-partition-array-into-two-equal-product-subsets/partition-array-into-two-equal-product-subsets.cpp
class Solution {
public:
    static bool checkEqualPartitions(vector<int>& nums, long long target) {
        const int n=nums.size(), bmask=(1<<n)-1;
        for(int mask=1; mask<=bmask; mask++){
            unsigned long long A[2]={1, 1};
            for(int i=0; i<n; i++){
                A[((mask>>i)&1)==1]*=nums[i];
            }
            if (A[0]==target && A[1]==target) return 1;
        }
        return 0;
    }
};

File: solutions/3843-partition-array-into-two-equal-product-subsets/partition-array-into-two-equal-product-subsets.java
class Solution {
    public boolean checkEqualPartitions(int[] nums, long target) {
        long p = 1;
        for(int i = 0;i<nums.length;i++){
            p = p*nums[i];
        }
       if(p != target*target){
        return false;
       }
       return helpr(target,nums.length,nums,1);
    }
    public boolean helpr(long target,int n,int[]nums,long prd){
    if(prd > target){
        return false;
    }
    if(target == prd){
        return true;
    }
    if(n == 0){
    return prd == target;
    }
    if(helpr(target,n-1,nums,prd*nums[n-1])){
        return true;
    }
    return helpr(target,n-1,nums,prd);
    }
}